cmd/uhc/cluster/list/list.go

```diff
@@ -85,23 +93,68 @@ func prettyPrint(arrStr ...[]string) {
 }
 
 // updateRowPad updates the length of all strings in a given list
-// to match wanted length
-func updateRowPad(columnPad []int, st []string) []string {
-	if len(columnPad) == len(st) {
-		for i := range st {
-			// Add padding
-			if len(st[i]) < columnPad[i] {
-				st[i] = st[i] + strings.Repeat(" ", columnPad[i]-len(st[i]))
-				// Clip
+// to match wanted length:
+func updateRowPad(columnPad []int, columnList []string) []string {
+	// Make sure padding list is at least as long as row list
+	st := columnList
+	fixLen := len(columnPad) - len(st)
+	if fixLen < 0 {
+		// Get last value of column pad and simply re-use it to fill columnPad up
+		valueToUse := columnPad[len(columnPad)-1]
+		for i := 0; i < fixLen*(-1); i++ {
+			columnPad = append(columnPad, valueToUse)
+		}
+	}
+	for i := range st {
+		// Add padding
+		if len(st[i]) < columnPad[i] {
+			st[i] = st[i] + strings.Repeat(" ", columnPad[i]-len(st[i]))
+			// Clip
+		} else {
+			st[i] = st[i][:columnPad[i]-2] + "  "
+		}
+	}
+	return st
+}
+
+// findMapValue will find a key and retrieve its value from a given map.
+// key has to be a string and can be multilayered i.e foo.bar
+// returns bool, true if key was found else false
+func findMapValue(data map[string]interface{}, str *string) bool {
+
+	// Split key into array
+	stringSlice := strings.Split(*str, ".")
+
+	// Create a local map with a pointer to it
+	mapData := data
+
+	// loop though elements in sliced string:
+	for _, element := range stringSlice {
+
+		// if key is found, continue:
+		if val, ok := mapData[element]; ok {
+
+			// If key points to string:
```

Comment:
When you need to check if a value has one of multiple types it is better to use a _type switch_:

```go
switch typed := val.(type) {
case string:
        value = typed
        ok = true
        return
case map[string]interface{}:
        data = typed
default:
        value = none
        return
}
```

I'd also suggest to include a `case` for integers, floating point numbers and dates, as some columns may be of those types.

Reply:
Added case for:
 - int
 - float32/634

from what I saw dates are strings so I made no changes there.